### PR TITLE
feat: `sshnp_register_tool/`

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
# What I did

Create a register tool for [sshnp](https://github.com/atsign-foundation/sshnoports). Easily get and onboard atSigns for the client and device and also easily create the sshnpd.service to later edit via `nano`.

Wrote [README.md](https://github.com/JeremyTubongbanua/at_tools/blob/trunk/sshnp_register_tool/README.md) instructions

### CODE REVIEW

**Comment by JeremyTubongbanua** [comment]
Just need to test on Windows & Linux, then will open this PR.

File: sshnp_register_tool/README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+## 1. Dart
+
+Fork the repository (or clone it) and run the tool from the repository itself via `dart run`.
+
+```sh
+dart run bin/sshnp_register_tool.dart -e <email@email.com>
+```
+
+## 2. Binary
```
Comment (cpswan) [inline sshnp_register_tool/README.md:20]: Can we create binaries (using a GitHub Action) and make them available here as releases?

File: sshnp_register_tool/README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+# sshnpd.service
+
+Selecting 'd' then option (3.) will initialize the `sshnpd.service`. This service will run the `sshnpd` daemon on your device when the device boots up. If the sshnpd.service file needs to be edited, just edit the file in `lib/` and the code will read straight from that file.
+
+Once the file is created, the tool will ask you to edit the file via `nano`. This is because line 11 needs to be changed to the atSigns that both the client and device will be using to communicate ssh without open ports.
+
+# Examples
+
+1. example of onboarding a free new atSign
```
Comment (cpswan) [inline sshnp_register_tool/README.md:49]: Each of the examples is quite long, and would benefit from using the details tag - https://gist.github.com/scmx/eca72d44afee0113ceb0349dd54a84a2

File: sshnp_register_tool/bin/sshnp_register_tool.dart [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+import 'package:args/args.dart';
+import 'package:sshnp_register_tool/colors.dart';
+import 'package:sshnp_register_tool/io_util.dart';
+import 'package:sshnp_register_tool/api_util.dart';
+import 'package:sshnp_register_tool/onboard_util.dart';
+import 'package:sshnp_register_tool/systemd_util.dart';
+
+/// Sets up your device's atSign for you. Just enter your email address and the binary will cycle through atSigns for you to choose from. Once you choose an atSign, it will prompt you for an OTP (one-time-password). Upon giving the correct OTP, it will activate your atSign and generate an .atKeys file inside the `~/.atsign/keys/` directory.
+/// This binary should realistically be only ran once to initialize and register the device's atSign.
```
Comment (cpswan) [inline sshnp_register_tool/bin/sshnp_register_tool.dart:12]: s/ran/run/

File: sshnp_register_tool/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+environment:
+  sdk: '>=2.18.0 <3.0.0'
+
+# dependencies:
+#   path: ^1.8.0
+
+dev_dependencies:
+  lints: ^2.0.0
+  test: ^1.16.0
+dependencies:
```
Comment (cpswan) [inline sshnp_register_tool/pubspec.yaml:15]: Please add a dependabot config to keep these up to date. There should also be tests to ensure that everything still works when a dependency is bumped, and when those tests pass we can build and release binaries.

**Review by cpswan — CHANGES_REQUESTED** [review]
Great work @JeremyTubongbanua 

One of my asks is trivial, and is a quick fix before merging. The other is more of a heavy lift, and I'm happy for it to be turned into issues for future work.

**Review by cconstab — COMMENTED** [review]
As this in its current format is specific to sshnoports perhaps it belongs in that repo.

I had imagined a standalone tool useful for anything.

Again perhaps that is a better thing to do here and have the sshnoports extensions in the sshnp repo..

Very cool in either case ty!

**Comment by gkc** [comment]
@JeremyTubongbanua Please can you spend a little time to address the comments in this PR?

**Comment by JeremyTubongbanua** [comment]
Hi, sorry, I will try to get on this next week when my capacity is larger.

**Comment by cconstab** [comment]
Still open!

**Pull request closed without merging**